zapcore/sampler.go

```diff
@@ -81,17 +81,87 @@ func (c *counter) IncCheckReset(t time.Time, tick time.Duration) uint64 {
 	return 1
 }
 
+// SamplingDecision represents a decision made by sampler.
```

Comment:
nit: all types represent something. can do:

```suggestion
// SamplingDecision is a decision made by sampler.
```